ocis/pkg/runtime/service/service.go

```diff
@@ -515,7 +515,8 @@ func trap(s *Service, ctx context.Context) {
 		}
 	}
 	s.Log.Debug().Str("service", "runtime service").Msgf("terminating with signal: %v", s)
-	os.Exit(0)
+	time.Sleep(3 * time.Second) // give the services time to deregister
+	os.Exit(0)                  // FIXME this cause an early exit that prevents services from shitting down properly
```

Comment:
It is still here btw :smile: but we can keep it as a reminder.

Reply:
Yes, I was thinking of removing the comment, but it feels right like it is. reva has no way to properly handle a context done triggered shutdown. A runtime.Shutdown(ctx) function like the http server would be great and would allow us to properly shut down all services ... currently we have no way of notifying decomposedfs to flush data to disk ...